store/src/java/com/zimbra/cs/service/mail/SearchResponse.java

```diff
@@ -149,13 +149,17 @@ int size() {
      * @param hit hit to append
      * @throws ServiceException error
      */
-    void add(ZimbraHit zimbraHit) throws ServiceException{
-		add(zimbraHit,false);
+    void add(ZimbraHit zimbraHit) throws ServiceException {
+        add(zimbraHit, false);
+    }
+
+    void add(ZimbraHit zimbraHit, boolean expandMsg) throws ServiceException {
+        add(zimbraHit, expandMsg, false);
+    }
```

Comment:
both look incorrect; a wrong function seems to be called on each.

Reply:
we have 3 add methods,  

> add(ZimbraHit zimbraHit)

> add(ZimbraHit zimbraHit, boolean expandMsg)

> add(ZimbraHit hit, boolean expandMsg, boolean isSharedFileWithMe)

if ` add(ZimbraHit zimbraHit)` is invoked then we assume `expandMsg` is false and similarly `isSharedFileWithMe` also becomes false.
similary when `add(ZimbraHit zimbraHit, boolean expandMsg) ` is called the, apart from respective values , `isSharedFileWithMe` becomes false.
And when `add(ZimbraHit hit, boolean expandMsg, boolean isSharedFileWithMe)` is called all the respective values are provided
May I know what exactly seems wrong here so I can fix it up ?